src/Storages/MergeTree/DataPartStorageOnDisk.cpp

```diff
@@ -121,6 +121,21 @@ static UInt64 calculateTotalSizeOnDiskImpl(const DiskPtr & disk, const String &
     return res;
 }
 
+UInt64 DataPartStorageOnDisk::calculateTotalSizeOnDisk(const DiskPtr & disk, const String & from)
+{
+    try
+    {
+        /// Files of detached part may be not exist, and then set file size is 0.
+        if (!disk->exists(from))
+            return 0;
+        return calculateTotalSizeOnDiskImpl(disk, from);
```

Comment:
let's remove try catch and exists check and do this in StorageSystemDetachedParts, because now the behaviour is inconsistent.